src/controllers/business.js

```diff
@@ -183,251 +185,492 @@ module.exports.uploadFile = (req, res) => {
+//           //   Bill.findOneAndUpdate(
+//           //     {
+//           //       clubMembershipId: bill.Membershipnumber,
+//           //       classId: req.body.classid,
+//           //       billDate: req.body.BillDate,
+//           //     },
+//           //     {
+//           //       $set: {
+//           //         paidAt: Date.now(),
+//           //       },
+//           //     },
+//           //     { new: true, useFindAndModify: false },
+//           //     (err) => {
+//           //       if (err) {
+//           //         console.log(err);
+//           //         return res.status(400).json({
+//           //           err: `Bill  updation failed !!! at line no ${index + 1}`,
+//           //         });
+//           //       }
+//           //     }
+//           //   );
+//           // });
+//           // return res.status(400).json({
+//           //   success: `Bill  updation done `,
+//           // });
+
+//           //**************************  if no errors present updating the whole bills from preadsheet data  */
+//           await Bill.bulkWrite(
+//             data.map((bill) => ({
+//               updateOne: {
+//                 filter: {
+//                   clubMembershipId: bill.Membershipnumber,
+//                   classId: req.body.classid,
+//                   billDate: req.body.BillDate,
+//                 },
+//                 update: {
+//                   $set: {
+//                     paidAt: Date.now(),
+//                   },
+//                 },
+//                 new: true,
+//                 useFindAndModify: false,
+//               },
+//             })) //,
+//             // {},
+//             // (err) => {
+//             //   if (err) {
+//             //     console.log(err);
+//             //     return res.status(400).json({
+//             //       err: `Bill  updation failed !!! at line no `,
+//             //     });
+//             //   }
+//             // }
+//           );
+//         }
+//       })
+//       //finally delete the uploaded spreadsheets from server
+//       .then(async () => {
+//         await unlinkAsync(`./temp/xlsx/${req.file.originalname}`);
+//         console.log("im here deleted spread sheet");
+//         return;
+//       });
+//     //console.log(data);
+//   });
+// };
 
-      cb(null, true);
-    },
-  }).single("payment");
-  upload(req, res, function (err) {
-    if (err) {
-      return res.json(err);
-    }
-    //**************************** */
-    //read xlsx file from folder
-    var workbook = XLSX.readFile(`./temp/xlsx/${req.file.originalname}`, {
-      type: "binary",
-      cellDates: true,
+module.exports.uploadXLXSFile = async (req, res) => {
+  // get xlsx workbook
+  var workbook = await storageXlsx.getWorkBook(req.file);
+  var sheet_name_list = workbook.SheetNames;
+  let data = [];
+
+  //converting xlxsworkbook to array of json data
+  xlsxToJson(sheet_name_list, workbook, data);
+
+  let Errors = [];
```

Comment:
Since Errors is not a class, It should not start with Capital.

rename it to something more readable. So that we could understand what data does the variable contains by reading the name itself.
e.g
`errors` or `errorsInData`.